megamek/src/megamek/common/MovePath.java

```diff
@@ -1341,10 +1369,10 @@ public boolean isMoveLegal() {
         }
 
         // for aero units move must use up all their velocity
-        if (getEntity().isAero()) {
-            IAero a = (IAero) getEntity();
+        // but only if it is actually IAero, because anything could return isAero() == true but not implement IAero
+        if (getEntity().isAero() && getEntity() instanceof IAero aero) {
```

Comment:
Do we know what returns isAero() but doesn't implement it? Feels like that's an oversight.

Reply:
I dont know, but I also dont want to risk casting to an interface without checking if it is castable